Rate limiting (for a zone)

benjvi opened the pull request:
Adds support for the rate limiting API. Mostly standard CRUD calls to API with marshalling etc. For listing, I allowed the caller to specify pagination options or call ListAll directly

jamesog commented on rate_limiting.go:
```diff
@@ -0,0 +1,194 @@
+
+import (
+	"encoding/json"
+	"github.com/pkg/errors"
```
Nit: the style elsewhere in this project (presumably because most people are using `goimports`) is to put external imports at the end, after a blank line.

jamesog commented on rate_limiting_example_test.go:
```diff
@@ -0,0 +1,109 @@
+
+import (
+	"fmt"
+	cloudflare "github.com/cloudflare/cloudflare-go"
```
Same here.

jamesog commented on rate_limiting.go:
```diff
@@ -0,0 +1,194 @@
+// RateLimit is a policy than can be applied to limit traffic within a customer domain
+type RateLimit struct {
+	ID          string                  `json:"id,omitempty"`
+	Disabled    bool                    `json:"disabled,omitempty"` // api defaults to false
```
Not sure if the comment is necessary here as bool also defaults to false

benjvi commented:
Thanks guys. I addressed the comments from @jamesog, also I realised today that the empty struct values were different than the api defaults when completely omitting a field in a couple of places -
 which results in the wrong thing happening when empty values are given. I replaced these with pointers so the values get omitted completely and default values are used. Hopefully this should be ok now?
